### DISCUSSION TITLE
"singularity: Command not found." after qsub

### DISCUSSION THREAD

**Discussion by ttameshige** [nig-sc/nigsc_homepage2#345] [Q&A] [answered]
遺伝研スーパーコンピュータを利用させて頂きありがとうございます。

初心者で何かと基本的なことを理解していない可能性があるので恐縮ですが、質問させてください。

まずは singularity　や　qsub の使い方を練習しようとしたのですが、
singularity　が使えないエラーが出て困っている状況です。

練習というのは、
以下のコマンドだけが書かれたファイル my_lolcow.sh を用意して
```
singularity run lolcow.sif
```

```
$ qsub my_lolcow.sh
```
を実行したのですが、エラー出力で

'singularity: Command not found.'

と書かれていました。qsub でジョブを投入する場合には singularity コマンドは使えないのでしょうか？

なおログインノード内でシンプルに以下を実行した際は正しくイラストが出力されました。
```
$ singularity run lolcow.sif
```

lolcow.sif は以下の遺伝研チュートリアルページにあるものです。
[Apptainer (Singularity)の使い方](https://sc.ddbj.nig.ac.jp/software/Apptainer/)

**Comment by kfuku52** [comment]
横から失礼します。`my_lolcow.sh`に`module load singularity/3.2.0`と加えたらsingularityを呼び出せるようにならないでしょうか。

**Reply by ttameshige** [reply]
なるほど。モジュールの呼び出しステップが必要なのですね。ありがとうございます！

**Comment by sc-fs** [comment]
qsubで実行されたmy_lolcow.shに、スクリプト実行時のインタプリタの指定（#$ -S /bin/bash）が無いようです。
一度、以下を参照して頂き、スクリプトに必要なパラメータを確認ください。
https://sc.ddbj.nig.ac.jp/software/grid_engine/batch_jobs

**Reply by ttameshige** [reply]
インタプリタの指定の必要性（および他の#$で始まるオプション指定）を理解していませんでした。ありがとうございました。

**Reply by ttameshige** [reply, marked as answer]
その後報告を失念しておりました。すみません。
以下のシェルスクリプトで期待通り動きました。sc-fs様、ご助言ありがとうございました。
```
#!/bin/bash
#$ -S /bin/bash
singularity run lolcow.sif
```
